exercises/all-your-base/src/test/java/BaseConverterTest.java

```diff
@@ -0,0 +1,256 @@
+                actualDigits);
+    }
+
+    @Test
+    public void test15BitInteger() {
+        final BaseConverter baseConverter = new BaseConverter(97, new int[]{3, 46, 60});
+
+        final int[] expectedDigits = new int[]{6, 10, 45};
+        final int[] actualDigits = baseConverter.convertToBase(73);
+
+        assertArrayEquals(
+                String.format(
+                        "Expected digits: %s but found digits: %s",
+                        Arrays.toString(expectedDigits),
+                        Arrays.toString(actualDigits)),
+                expectedDigits,
+                actualDigits);
+    }
+
+    @Test
+    public void testEmptyDigits() {
+        expectedException.expect(IllegalArgumentException.class);
+        expectedException.expectMessage("You must supply at least one digit.");
+
+        new BaseConverter(2, new int[]{});
+    }
+
+    @Test
+    public void testSingleZero() {
+        final BaseConverter baseConverter = new BaseConverter(10, new int[]{0});
+
+        final int[] expectedDigits = new int[]{0};
+        final int[] actualDigits = baseConverter.convertToBase(2);
+
+        assertArrayEquals(
+                String.format(
+                        "Expected digits: %s but found digits: %s",
+                        Arrays.toString(expectedDigits),
+                        Arrays.toString(actualDigits)),
+                expectedDigits,
+                actualDigits);
+    }
+
+    @Test
+    public void testMultipleZeros() {
+        expectedException.expect(IllegalArgumentException.class);
+        expectedException.expectMessage("Digits may not contain leading zeros.");
+
+        new BaseConverter(10, new int[]{0, 0, 0});
+    }
+
+    @Test
+    public void testLeadingZeros() {
+        expectedException.expect(IllegalArgumentException.class);
+        expectedException.expectMessage("Digits may not contain leading zeros.");
+
+        new BaseConverter(7, new int[]{0, 6, 0});
+    }
+
+    @Test
+    public void testNegativeDigit() {
+        expectedException.expect(IllegalArgumentException.class);
+        expectedException.expectMessage("Digits may not be negative.");
+
+        new BaseConverter(2, new int[]{1, -1, 1, 0, 1, 0});
+    }
+
+    @Test
+    public void testInvalidPositiveDigit() {
+        expectedException.expect(IllegalArgumentException.class);
+        expectedException.expectMessage("All digits must be strictly less than the base.");
+
+        new BaseConverter(2, new int[]{1, 2, 1, 0, 1, 0});
+    }
+
+    @Test
+    public void testFirstBaseIsOne() {
+        expectedException.expect(IllegalArgumentException.class);
+        expectedException.expectMessage("Bases must be at least 2.");
+
+        new BaseConverter(1, new int[]{});
+    }
+
+    @Test
+    public void testSecondBaseIsOne() {
+        final BaseConverter baseConverter = new BaseConverter(2, new int[]{1, 0, 1, 0, 1, 0});
+
+        expectedException.expect(IllegalArgumentException.class);
```

Comment:
All this time, I had anchor bias and put my `ExpectedException` rules at the top of the method (prior to the rule, we had the annotation parameter).  But you are putting it immediately before the method that should emit the exception... I ❤️  this.